be/src/exec/vectorized/sorting/merge_column.cpp

```diff
@@ -304,6 +316,31 @@ ChunkUniquePtr SortedRun::clone_slice() const {
     }
 }
 
+ChunkPtr SortedRun::steal_chunk(size_t size) {
```

Comment:
copy or steal?

Reply:
This method is used to reduce the memory copy of chunk. The expected scenario is just returning the chunk in the `SortedRun` without any copy.

As LocalMerge scenario, all chunks are merged into `SortedRuns`, which consists of many ordered chunks. For output, there's a restriction that the chunk size should not greater than `state->chunk_size()`.  So if the chunk in `SortedRuns` meet the requirement, we could `steal` the chunk from `SortedRuns` directly without any copy, but if not, we need to chop the chunk into smaller size, that would introduce extra overhead.

Reply:
In general, the overhead of merge procedure is at memory copy. Current implementation is not optimal, and needs some extra engineering effort.